control-plane/agents/src/bin/core/tests/pool/mod.rs

```diff
@@ -1130,3 +1134,264 @@ async fn slow_create() {
+        client
+            .pool()
+            .destroy(&destroy, None)
+            .await
+            .expect_err("Pool-2 destruction should fail");
+
+        client
+            .pool()
+            .create(&create_pool_2_request, None)
+            .await
+            .expect("Pool-2 creation should succeed now");
+
+        cluster
+            .composer()
+            .restart(&cluster.node(0))
+            .await
+            .expect("Cluster restart failed");
+        cluster
+            .wait_node_status(cluster.node(0), NodeStatus::Online)
+            .await
+            .expect("Node service liveness check failed");
+
+        let create_pool_3_request = CreatePool {
+            node: cluster.node(0),
+            id: "pool-3".into(),
+            disks: vec![format!(
+                "aio://{}",
+                device_devlinks
+                    .first()
+                    .expect("At least one devlink should be present")
+            )
+            .into()],
+            labels: None,
+            encryption: None,
+        };
+
+        client
+            .pool()
+            .create(&create_pool_3_request, None)
+            .await
+            .expect_err("Creating pool-3 with pool-1 devlink should fail");
+
+        client
+            .pool()
+            .destroy(&DestroyPool::from(create_pool_2_request.clone()), None)
+            .await
+            .expect("Destroying pool-2 should succeed");
+
+        client
+            .pool()
+            .destroy(&DestroyPool::from(create_pool_3_request.clone()), None)
+            .await
+            .expect_err("Destroying pool-3 should fail");
+    }
+
+    pub struct SecretFileCreator {
+        file_path: Option<PathBuf>,
+    }
+
+    impl SecretFileCreator {
+        pub async fn new() -> std::io::Result<Self> {
+            let root = env::var("WORKSPACE_ROOT").expect("WORKSPACE_ROOT is not set");
+            let path = PathBuf::from(&root).join(".tmp");
+
+            let data = json!({
+                "cipher": "AesXts",
+                "key": "2b7e151628aed2a6abf7158809cf4f3c",
+                "key_len": 128,
+                "key2": "2b7e151628aed2a6abf7158809cf4f3d",
+                "key2_len": 128
+            });
+
+            let file_path = path.join(SECRETFILE);
+            let mut file = File::create(&file_path).await?;
+            file.write_all(data.to_string().as_bytes()).await?;
+
+            Ok(Self {
+                file_path: Some(file_path),
+            })
+        }
+    }
+
+    impl Drop for SecretFileCreator {
+        fn drop(&mut self) {
+            if let Some(ref path) = self.file_path {
+                let path = path.clone();
+                std::thread::spawn(move || {
+                    let _ = std::fs::remove_file(path);
+                    println!("Removed secretfile");
+                });
+            }
+        }
+    }
+}
+
+#[tokio::test]
```

Comment:
Doesn't need to be async
Also since this is unittest, add it where `DEVLINK_REGEX`  lives

Reply:
Done.